config/management/operational/src/test_helper.rs

```diff
@@ -83,6 +83,30 @@ impl OperationalTool {
         command.check_endpoint()
     }
 
+    pub fn check_endpoint_with_key(
+        &self,
+        network_id: &NetworkId,
+        network_address: NetworkAddress,
+        private_key: &x25519::PrivateKey,
+    ) -> Result<String, Error> {
+        let args = format!(
+            "
+                {command}
+                --address {network_address}
+                --chain-id {chain_id}
+                --network-id {network_id}
+                --private-key {private_key}
+            ",
+            command = command(TOOL_NAME, CommandName::CheckEndpoint),
+            chain_id = self.chain_id.id(),
+            network_address = network_address,
+            network_id = network_id,
+            private_key = hex::encode_upper(private_key.to_bytes()),
```

Comment:
would be nice to just have a wrapper container / trait that does this for us something like .to_string_dangerously()

Reply:
i thought Debug was impl'ed in debug mode or did we just remove it altogether?

Reply:
I think both Display and Debug are implemented to print "<Redacted>" or some such

Reply:
> Its sort of funny to me that we go out of the way to avoid being able to print a private key and then we do this :P

Well, how else am I going to have a procedurally generated key in a test plugged into a CLI tool :P

> Also, why encode_upper?

Upper, because we use uppercase for most of our identifiers in hex.

> would be nice to just have a wrapper container / trait that does this for us something like .to_string_dangerously()

We could have a trait for printing to hex, but as brandon mentioned earlier we avoid printing private keys as much as possible.

> I think both Display and Debug are implemented to print "Redacted" or some such

Reply:
the standard way to serialize keys:

```rust
use diem_crypto::traits::ValidCryptoMaterialStringExt;

let private_key: x25519::PrivateKey = /* .. */;
let encoded = private_key.to_encoded_string()?;
```

Reply:
Yeah lets do this ^